### PR TITLE
[next] support sectioned source maps

### PR DESCRIPTION
**Pull request by wbinnssmith**
Turbopack outputs sectioned source maps, but webpack’s `webpack-sources` packages doesn’t support these. For performance reasons, it should, but as an interim solution, flatten these sections maps for compatibility.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,11 @@
     "turbo": "1.13.2",
     "typescript": "4.9.5"
   },
+  "pnpm": {
+    "overrides": {
+      "source-map": "0.7.4"
```
Comment (TooTallNate) [inline package.json]: Can you explain why this is necessary?

File: .changeset/smart-beds-sit.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@vercel/redwood': patch
+'@vercel/next': patch
+'@vercel/node': patch
+'api': patch
```
Comment (TooTallNate) [inline .changeset/smart-beds-sit.md]: The `api` package does not need to be versioned.

```suggestion
```

File: .changeset/smart-beds-sit.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+'@vercel/redwood': patch
+'@vercel/next': patch
+'@vercel/node': patch
```
Comment (TooTallNate) [inline .changeset/smart-beds-sit.md]: Change seems unrelated to redwood / node?

```suggestion
'@vercel/next': patch
```

File: packages/next/build.mjs [hunk 4]
```diff
@@ -18,6 +23,11 @@ await Promise.all([
     ],
   }),
   buildEdgeFunctionTemplate(),
+  // The bundled version of source-map looks for the wasm mappings file as a sibling.
+  copyFile(
+    fileURLToPath(import.meta.resolve('source-map/lib/mappings.wasm')),
```
Comment (TooTallNate) [inline packages/next/build.mjs]: `copyFile()` works with URLs:

```suggestion
   import.meta.resolve('source-map/lib/mappings.wasm'),
```

File: package.json (hunk 1 above)
Comment (wbinnssmith) [reply package.json]: It's not. reverted.

File: packages/next/build.mjs (hunk 4 above)
Comment (wbinnssmith) [reply packages/next/build.mjs]: It didn't for me. I got an `ENOENT` on the source file.

**Review by TooTallNate — APPROVED** [review]
